src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -83,7 +91,8 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
 
             if (!string.IsNullOrEmpty(ApplicationUri))
             {
-                ApplicationConfigUri = ApplicationUri + ConfigExtension;
+                string additionalExtension = isSingleFile ? ".dll" : null;
```

Comment:
I suspect that it is still not going to work for the repro attached to the bug on Windows. `currentProcess.MainModule?.FileName` above will return file name with `.exe` (e.g. `AppSetting.exe`). We will end up with `AppSetting.exe.dll` after appending `.dll` here and eventually look for `AppSetting.exe.dll.config`, but the actual file name is `AppSetting.dll.config`.

Reply:
Does it mean that we broken on Windows for all hosting forms? because it will produce `AppSetting.exe.config` with the current (master) revision. I was working on macOS.

Reply:
I have made the fix, it continues to work on macOS and Linux. Also booted up Windows machine and verified that FDD as well as SingleFile publish produce `AppSetting.dll.config`, while the existing code was looking for `AppSetting.exe.config`.